packages/components/src/components/primitives/Snackbar/Snackbar.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect } from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import { getStyle } from './Snackbar.styles';
+import {
+  Typography,
+  TypographyMode,
+  TypographyTypes,
+} from 'components/primitives/Typography';
+import { Icon } from 'components/primitives/Icon';
+
+export const SNACKBAR_DEFAULT_TEST_ID = 'snackbar';
+
+export enum SnackbarType {
+  Default = 'default',
+  Success = 'success',
+  Warning = 'warning',
+  Error = 'error',
+}
+
+export enum SnackbarTextPosition {
+  Left = 'left',
+  Right = 'right',
+}
+
+export enum SnackbarPosition {
+  TopCenter = 'top-center',
+  TopRight = 'top-right',
+  TopLeft = 'top-left',
+  BottomRight = 'bottom-right',
+  BottomCenter = 'bottom-center',
+  BottomLeft = 'bottom-left',
+}
+
+type Props = {
+  testId?: string;
+  open?: boolean;
+  message: string;
+  description?: string;
+  showIcon?: boolean;
+  icon?: React.ReactNode;
+  textPosition?: SnackbarTextPosition;
+  showButton?: boolean;
```

Comment:
is this the close button?

Reply:
It's a regular button. I've added a `Button` component and a new prop `onClickButton`

Reply:
what do you mean by a regular button?

Reply:
Changed implementation. As we agreed, now Snackbar accepts an optional action object.